tools/rum-models-generator/Sources/RUMModelsGeneratorCore/RUM/RUMSwiftTypeTransformer.swift

```diff
@@ -152,7 +161,8 @@ internal class RUMSwiftTypeTransformer: TypeTransformer<SwiftType> {
 
         // If the type name collides with Swift `Type` keyword, prefix it with parent type name.
         if fixedName == "Type" {
-            let parentTypeName = (context.parent as? SwiftStruct)?.name ?? ""
+            let parentStruct = context.predecessor(matching: { $0 is SwiftStruct }) as? SwiftStruct
+            let parentTypeName = parentStruct?.name ?? ""
```

Comment:
This is to fix the problem with [`[Frustration.Type]`](https://github.com/DataDog/rum-events-format/blob/d5a943ad55117da57d0b87f9af417ccbcbd49daf/schemas/rum/action-schema.json#L85) where `Type` cannot be an `enum` name (it conflicts with `Swift.Type`). This transformer already implements the logic of prefixing `Type` with parent `struct's` name, but in case of `[Frustration.Type]` the parent is an array. To fix this, I added `predecessor(matching:)` method which finds the transitive parent matching given predicate.